projects/igniteui-angular/src/lib/calendar/calendar.component.ts

```diff
@@ -37,6 +36,25 @@ export enum CalendarSelection {
     RANGE = 'range'
 }
 
+export enum DateRangeType {
+    After,
+    Before,
+    Between,
+    Specific,
+    Weekdays,
+    Weekends
+}
+
+export class DateRangeDescriptor {
```

Comment:
You don't need a class here. `interface` is sufficient, since all you do in the constructor is assign the two properties.